### PR TITLE
4413 civ 2.5 item ledger 

### LINKED ISSUE #4413: CIV 2.5: Item Ledger
The ledger is currently being shown in `View Stock` -> `Stockline` -> `Ledger`. We want to show the item ledger on the item itself as well. Bring in additional columns shown in OG. Can be broken down into two PRs or two separate issues for backend and frontend.

### Backend
Implement the following columns in the ledger view:
- status
- expiry
- batch
- pack_size
- balance 
	- previous balance + adjustment from invoice line
- cost_price (per pack)
- sell_price (per pack)
- local_currency_total (total_price_before_tax in local currency)

Frontend implementation:
- New tab in the `Item` detail view
- Tab should show ledger with pagination
- Add an on-click on the line that takes you to the relevant transaction

### Moneyworks Jobcode 🧰
OMS:CIDV

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #4413

# 👩🏻‍💻 What does this PR do?
- Returns extra field in the stock movement view
- Calculate the balance for the ledger row (by going back to the beginning and summing all records until the current record). 
- Item ledger endpoint
- Item ledger tab in the Item page that shows all the movements associated with that item
- Clicking on a line will redirect you to the invoice. Note: Inventory adjustment transactions will not redirect you anywhere
- Pagination for the item ledger table

item_ledger

## 💌 Any notes for the reviewer?
Clicking on an inventory adjustment line will not redirect the user to any transaction. Inventory adjustments can be created either through:
- Creating a new stock line from the stock page
- Editing a stock line from the stock page
- Through stocktakes
There is currently no way to tell what created the IA invoice.

# 🧪 Testing

- [ ] Click on `Item`
- [ ] Click `Ledger` tab
- [ ] See stock movements associated with that item
- [ ] Click on a line that is not an inventory adjustment
- [ ] Be redirect to the invoice

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditTabs.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const StocktakeLineEditTabs: FC<
             color="primary"
             variant="outlined"
             onClick={onAddLine}
-            label={`${t('label.add-batch', { ns: 'inventory' })} (+)`}
```
Comment (roxy-dao) [inline client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditTabs.tsx:68]: Sidequest: We don't namespace anymore

File: client/packages/system/src/Stock/Components/Ledger/useLedgerColumns.ts [hunk 2]
```diff
@@ -87,24 +87,3 @@ export const useLedgerColumns = (
 
   return { columns };
 };
-
-const getLocalisationKey = (type: InvoiceNodeType): LocaleKey => {
-  switch (type) {
```
Comment (roxy-dao) [inline client/packages/system/src/Stock/Components/Ledger/useLedgerColumns.ts:92]: Moved to a more common place to use in item ledger as well

File: client/packages/system/src/Item/DetailView/DetailView.tsx [hunk 3]
```diff
@@ -30,6 +32,53 @@ export const ItemDetailView: FC = () => {
...
+        navigate(
+          RouteBuilder.create(AppRoute.Dispensary)
+            .addPart(AppRoute.Prescription)
+            .addPart(String(ledger.invoiceNumber))
+            .build()
+        );
+        break;
+      default:
+        break;
+    }
```
Comment (fergie-nz) [inline client/packages/system/src/Item/DetailView/DetailView.tsx:79]: I'm wondering if toast / snack would be helpful where no navigation is available for inventory adjustments if it isn't obvious for users.

File: server/service/src/ledger.rs [hunk 4]
```diff
@@ -1,26 +1,80 @@
...
+    pagination: Option<PaginationOption>,
     filter: Option<LedgerFilter>,
     sort: Option<LedgerSort>,
 ) -> Result<ListResult<LedgerRow>, ListError> {
+    let pagination = get_default_pagination_unlimited(pagination);
     let connection = connection_manager.connection()?;
     let repository = LedgerRepository::new(&connection);
 
-    let rows = repository.query(Pagination::all(), filter.clone(), sort)?;
+    let rows = repository.query(pagination, filter, sort)?;
```
Comment (fergie-nz) [inline server/service/src/ledger.rs:25]: I'm curious about querying for everything here? I understand you need it for balance calculations, but would there be performance concerns for an item which has many adjustments?

File: client/packages/system/src/Item/DetailView/DetailView.tsx (hunk 3 above)
Comment (roxy-dao) [reply client/packages/system/src/Item/DetailView/DetailView.tsx:79]: I think okay for now. Something to note in the future maybe. This was a bonus and isn't in the specs XD

Comment (roxy-dao) [reply client/packages/system/src/Item/DetailView/DetailView.tsx:79]: Just not being able to click is fine. This was an added bonus, I don't know if they will actually use it

File: server/service/src/ledger.rs (hunk 4 above)
Comment (roxy-dao) [reply server/service/src/ledger.rs:25]: This is for the stock line which hopefully shouldn't have too many entries, but if it does then the pagination can be implemented easily (:

**Review by fergie-nz — APPROVED** [review]
Looks awesome to me, nice one Roxy

**Pull request merged**